dcicutils/redis_utils.py

```diff
@@ -59,13 +60,17 @@ def info(self) -> dict:
         """
         return self.redis.info()
 
-    def set(self, key: str, value: Union[str, int, float]) -> str:
+    def set(self, key: str, value: Union[str, int, float], exp: Union[int, datetime.timedelta] = None) -> str:
         """ Sets the given key to the given value https://redis.io/commands/set/
         :param key: string to store value under
         :param value: value mapped to from key
+        :param exp: expiration time in seconds as int or datetime.timedelta (optional)
         :return: a "true" string <OK>
         """
-        return self.redis.set(self._encode_value(key), self._encode_value(value))
+        kwargs = {}
+        if exp:
+            kwargs['ex'] = exp
+        return self.redis.set(self._encode_value(key), self._encode_value(value), **kwargs)
```

Comment:
Why using kwargs here instead of just `ex=exp` in redis.set call? (Nothing wrong with that just wondering).

Reply:
Just future proofing it in case additional kwargs are needed in the future